sklearn/cluster/tests/test_k_means.py

```diff
@@ -761,19 +717,19 @@ def test_float_precision(Estimator, data):
     assert_array_equal(labels[np.float32], labels[np.float64])
 
 
-def test_k_means_init_centers():
-    # This test is used to check KMeans won't mutate the user provided input
-    # array silently even if input data and init centers have the same type
-    X_small = np.array([[1.1, 1.1], [-7.5, -7.5], [-1.1, -1.1], [7.5, 7.5]])
-    init_centers = np.array([[0.0, 0.0], [5.0, 5.0], [-5.0, -5.0]])
-    for dtype in [np.int32, np.int64, np.float32, np.float64]:
-        X_test = dtype(X_small)
-        init_centers_test = dtype(init_centers)
-        assert_array_equal(init_centers, init_centers_test)
-        km = KMeans(init=init_centers_test, n_clusters=3, n_init=1)
-        km.fit(X_test)
-        assert np.may_share_memory(km.cluster_centers_,
-                                   init_centers) is False
```

Comment:
There's a mistake here. It should be testing against init_centers_test and not init_centers because KMeans is instanciated with init=init_centers_test.